cmd/nitro/init.go

```diff
@@ -205,7 +205,34 @@ func openInitializeChainDb(ctx context.Context, stack *node.Node, config *NodeCo
 						return chainDb, l2BlockChain, fmt.Errorf("failed to recreate missing states: %w", err)
 					}
 				}
-
+				latestBlock := l2BlockChain.CurrentBlock()
+				if latestBlock.Number.Uint64() <= chainConfig.ArbitrumChainParams.GenesisBlockNum {
+					// If there is only genesis block or no blocks in the blockchain, set Rebuilding of wasm store to Done
+					log.Info("setting rebuilding of wasm store to done")
+					if err = gethexec.WriteToKeyValueStore(wasmDb, gethexec.RebuildingPositionKey, gethexec.RebuildingDone); err != nil {
```

Comment:
another case to set it to done: if currentBlock has ArbOS version < params.ArbosVersion_Stylus

Reply:
(this is important so nodes that start with arb1/sepolia from before stylus upgrade won't spend time on rebuilding)

Reply:
I agree, This is a great catch!